File: Dataset/Leetcode/train/121/259.py
 class Solution:
    def XXX(self, prices: List[int]) -> int:
        if not prices:return 0
        profit = 0
        min_prices = prices[0]
        for i in prices:
            profit = i - min_prices if i - min_prices > profit else profit
            min_prices = i if i < min_prices else min_prices
        return profit

File: Dataset/Leetcode/valid/83/259.java
class Solution {
    public ListNode XXX(ListNode head) {
        // 既然是升序，那就以一个节点为主
        if (head == null || head.next == null) {
            return head;
        }
        ListNode cur = head;
        ListNode node = head.next;
        while(node != null) {
            while (node != null && node.val == cur.val) {
                node = node.next;
            } 
            if (node != null) {
                // 拼接在cur的后面
                ListNode tmp = node.next;

                cur.next = node;
                cur = node;
                node.next = null;

                node = tmp;
            }
        }
        cur.next = null;
        return head;
    }
}

